forge/forge/tvm_to_python.py

```diff
@@ -2735,40 +2750,79 @@ def generate_op_tests(
     including headers, class definitions, forward functions, parameter parsers, and pytest functions.
     The generated tests are designed to run the operations as standalone tests.
     """
-    for op_idx, key in enumerate(sorted(ops)):
+    for op_idx, nid in enumerate(sorted(ops)):
+
         # Create unique module name
         module_name = "test_" + current_module_name.lower() + str(op_idx)
 
-        # Initialize Forge writer and generate header and class definition
+        # Initialize Forge writer and generate header
         writer = ForgeWriter(
             module_name,
             framework,
             contains_incompatible_np_floats=contains_incompatible_np_floats,
             delete_inputs=delete_inputs,
         )
         writer.write_header()
-        writer.write_class_definition(params, constants)
-
-        # Focus on generating test for a single op
-        single_op = {key: ops[key]}
-
-        # Create new inputs for the single op
-        new_inputs = {}
-        for i, input_name in enumerate(single_op[key].input_names):
-            # Detected parameter as input, insert dummy input
-            # TODO: Need to handle this case better. Probably just ignoring
-            # model parameters, and using new generated inputs.
-            if "." in input_name:
-                input_name = "dummy_input_" + str(i)
-            new_inputs[input_name] = input_name
 
-        # Force output to be same as the op we're running
-        single_return = {key: single_op[key].output_name}
-
-        # Generate forward function and parameter parser (loading params and constants)
-        writer.write_forward(single_op, new_inputs, single_return)
-        writer.write_param_parser(param_names, param_file_name, names_params_file_name, named_buffers_file_name)
+        # Generating test for a single op
+        single_op = {nid: ops[nid]}
+
+        forward_method_inputs = {}
+        test_inputs_shapes = []
+        test_inputs_dtypes = []
+        activation_param_names = []
+        needed_params = {}
+        needed_consts = {}
+        
+        # Lambda function to check whether the input for the single op is const or parameter based upon the input names.
+        is_params_const = lambda inp_name: True if (inp_name.startswith("const") or inp_name.endswith(".weight") or "." in inp_name) else False
```

Comment:
Next to checking name, can we have more robust way of determining if type is param or constant?

Reply:
In the Operation class itself, I will add additional data member about the operand type info like Parameter/Constant/Activation by comparing the inputs node id with params and const dict(Key: node id, value: tuple(param_name, param_shape, require_grad, datatype))